solver/src/liquidity/offchain_orderbook.rs

```diff
@@ -15,23 +15,24 @@ impl OrderBookApi {
             .await
             .context("failed to get orderbook")?
             .into_iter()
-            .map(|order| order.order_creation.into())
+            .map(|order| order.into())
             .collect())
     }
 }
 
-impl From<OrderCreation> for LimitOrder {
-    fn from(order: OrderCreation) -> Self {
+impl From<Order> for LimitOrder {
+    fn from(order: Order) -> Self {
```

Comment:
I'd keep this as OrderCreation and use OrderCreation::uid() to get the id.

Reply:
The issue is that `uid()` needs the owner (which is also on metadata). We could recover owner from signature but that would require knowledge of the domain_separator here.

Reply:
I forgot it needed the owner. Makes sense to use the full order struct in that case.